components/post_view/post_message_view/system_message_helpers.jsx

```diff
@@ -63,13 +75,29 @@ function renderAddToChannelMessage(post, options) {
     );
 }
 
+function renderAddToTeamMessage(post, options) {
+    const username = renderUsername(post.props.username, options);
+    const addedUsername = renderUsername(post.props.addedUsername, options);
+
+    return (
+        <FormattedMessage
+            id='api.team.add_member.added'
+            defaultMessage='{addedUsername} added to the team by {username}'
+            values={{
+                username,
+                addedUsername
+            }}
+        />
+    );
+}
+
 function renderRemoveFromChannelMessage(post, options) {
     const removedUsername = renderUsername(post.props.removedUsername, options);
 
     return (
         <FormattedMessage
             id='api.channel.remove_member.removed'
-            defaultMessage='{removedUsername} was removed from the channel'
+            defaultMessage='{removedUsername} was removed from the team'
```

Comment:
This should still say `channel` when the user is just being removed from a random channel. It should only say `team` when they are actually removed from the entire team (i.e. in `town-square`).